shell/platform/embedder/embedder.cc

```diff
@@ -1253,6 +1253,182 @@ void PopulateAOTSnapshotMappingCallbacks(
   }
 }
 
+// Translates engine semantic nodes to embedder semantic nodes.
+FlutterSemanticsNode CreateEmbedderSemanticsNode(
+    const flutter::SemanticsNode& node) {
+  SkMatrix transform = node.transform.asM33();
+  FlutterTransformation flutter_transform{
+      transform.get(SkMatrix::kMScaleX), transform.get(SkMatrix::kMSkewX),
+      transform.get(SkMatrix::kMTransX), transform.get(SkMatrix::kMSkewY),
+      transform.get(SkMatrix::kMScaleY), transform.get(SkMatrix::kMTransY),
+      transform.get(SkMatrix::kMPersp0), transform.get(SkMatrix::kMPersp1),
+      transform.get(SkMatrix::kMPersp2)};
```

Comment:
This matrix is actually defined on the original semantics node as [a 4x4 matrix](https://github.com/flutter/engine/blob/2a1be6d4db1604004ee6d5dee1c70393948d67ba/lib/ui/semantics/semantics_node.h#L138). I wonder if we should consider exposing that if we're adding a cleaned-up API. The iOS/Android embedders consume it as 4x4.

That does have the downside of making migration from the old API to the new one more annoying for existing embedders. I'm also not sure that there are many a11y subsystems out there that support skewed or non-2D a11y nodes. 

In the end, we're always converting to screen rects in any case. e.g.
https://github.com/flutter/engine/blob/2a1be6d4db1604004ee6d5dee1c70393948d67ba/shell/platform/darwin/ios/framework/Source/SemanticsObject.mm#L38-L92

Thoughts?

Reply:
Thinking about this, let's wait until someone provides a strong use-case for this. That change is also orthogonal to this one since the way we'd add support for a 4x4 matrix is that we'd just use the same semantics node struct and add the additional class (`FlutterTransform44`) and field to that struct.

Reply:
Thanks for bringing this up, I wasn't aware of this! I agree this change can be done later if someone provides a strong use-case. With some luck we'll have a better deprecation policy then so that the embedder API doesn't get too gross!